### PR TITLE
fixed the naming for Sass/Scss in documentation, tests, formats

### PR DESCRIPTION
**Pull request by didoo**
*Issue #, if available:*
#213

*Description of changes:*
I have renamed the `sass/map-***` formats to `scss/map-***` and updated some documentation. 

We have to consider that this is technically a breaking change, so it's better to consider how to release it. One possibility is to include it in the next "major" release (3.0). One other is to release it as "minor", but show an error message for the users that are using the "old" format name (I don't know if it's possible, maybe someone has better ideas? I'm open to suggestions)

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by dbanksdesign** [comment]
Instead of having a deprecation warning for 'sass/map-*' formats, could we have a warning that those formats have moved to 'scss/map-*' and say the sass formats will now be sass syntax? I still need to figure out how to do maps in sass syntax...

**Comment by didoo** [comment]
@dbanksdesign I have made this very rough test: I have extracted from https://adele.uxpin.com/ the design systems that declare are using Sass (filter in the column "CSS"); they are around 45.

I have then checked for all of them which syntax flavor they are using `sass` or `scss` and here are the results: https://docs.google.com/spreadsheets/d/17KYJwxR1hbfkT60Isq9v7ZOlPYZZBe7kjimLGsD1acQ/edit?usp=sharing

As you can see (and as I suspected) out of 40 design systems only 1 is using `sass` syntax, all the others are using `scss`. Why this exercise? To justify why, for me, we should avoid having/supporting a double format in Style Dictionary. Because almost everyone is using `scss`, and because having both would lead to inevitable confusion (how many would use `sass` when they actually mean `scss`, like I did?). I know is not statistically accurate, but I think this is a strong indicator for a KISS approach here.

File: index.js [hunk 1]
```diff
@@ -113,4 +114,28 @@ custom templates:
   ${register_template_warnings}`));
   }
 
+  if(GroupMessages.count(SASS_MAP_FORMAT_DEPRECATION_WARNINGS) > 0) {
+    var sass_map_format_warnings = GroupMessages.flush(SASS_MAP_FORMAT_DEPRECATION_WARNINGS).join('\n  ');
+    console.log(chalk.bold.yellow(`
+⚠️ DEPRECATION WARNING ️️️️️⚠️
+The formats 'sass/map-***' have been renamed to 'scss/map-***', please update your config accordingly (will be soon removed).
```
[the remark this answers is not in the document]
Comment (chazzmoney) [reply index.js]: Not sure what the exact right message is - @dbanksdesign thoughts?

[the remark this answers is not in the document]
Comment (didoo) [reply index.js]: @chazzmoney I like your message. if it's ok for @dbanksdesign I'll update it

**Comment by didoo** [comment]
> I would lean towards including more if they are about language syntax, even if they are less popular languages. If it helps 2 people to not have to write their own format, I would say that is a win.
Well, it depends: 

- how much is the (future) cost of maintenance of that extra feature? (imagine the impact of a refactoring, of fixing failing tests, etc)? 
- how much complexity does it add to the overall codebase/project? 
- how much makes things less clear for a new user?
- etc
- 

I tend to apply the Pareto principle in this kind of cases ;)

(also, I have seen too many open-source projects explode (and maintainers go in burnout) because of feature creep).

File: index.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (dbanksdesign) [reply index.js]: Lets do it!

**Comment by dbanksdesign** [comment]
Very true. I have talked to some people about the idea of making style dictionary more like webpack, where people can publish extensions/plugins (or in our case transforms/formats/actions) so someone could publish a sass-style-dictionary package that adds sass specific transforms/formats. Then we wouldn't really need to worry about what we include in the core library as much. And actually, the node example PR is the first step to make that happen...

**Comment by didoo** [comment]
I have updated the message. This is how it looks like in the console now:
screenshot_2249

**Comment by chazzmoney** [comment]
@didoo I merged a bunch of things and the custom-format change caused a small conflict here. Can you update this one? Then we can merge.

**Comment by didoo** [comment]
@chazzmoney strange, I pulled upstream/develop and merged it into my branch, and there were no conflicts. can you check if everything is OK now?

**Comment by dbanksdesign** [comment]
Looks good to me!

**Pull request merged**